src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -378,61 +368,52 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
                          real64 const & fluidDensity,
                          real64 const & dFluidDensity_dPressure,
-                         real64 const & porosityInit,
                          real64 const & porosity,
                          real64 const & dPorosity_dVolStrain,
                          real64 const & dPorosity_dPressure,
                          real64 const ( &gravityVector )[3],
-                         real64 ( & bodyForceIncrement )[3],
+                         real64 ( & bodyForce )[3],
                          real64 ( & dBodyForce_dVolStrainIncrement )[3],
                          real64 ( & dBodyForce_dPressure )[3] ) const
   {
     real64 const mixtureDensity = ( 1.0 - porosity ) * solidDensity
                                   + porosity * fluidDensity;
-    real64 const initialMixtureDensity = ( 1.0 - porosityInit ) * solidDensity
-                                         + porosityInit * initialFluidDensity;
-    real64 const mixtureDensityIncrement = mixtureDensity - initialMixtureDensity;
 
     real64 const dMixtureDens_dVolStrainIncrement = dPorosity_dVolStrain * ( -solidDensity + fluidDensity );
     real64 const dMixtureDens_dPressure = dPorosity_dPressure * ( -solidDensity + fluidDensity )
                                           + ( 1.0 - porosity ) * m_porosityUpdate.dGrainDensity_dPressure()
                                           + porosity * dFluidDensity_dPressure;
 
-    LvArray::tensorOps::scaledCopy< 3 >( bodyForceIncrement, gravityVector, mixtureDensityIncrement );
+    LvArray::tensorOps::scaledCopy< 3 >( bodyForce, gravityVector, mixtureDensity );
     LvArray::tensorOps::scaledCopy< 3 >( dBodyForce_dVolStrainIncrement, gravityVector, dMixtureDens_dVolStrainIncrement );
     LvArray::tensorOps::scaledCopy< 3 >( dBodyForce_dPressure, gravityVector, dMixtureDens_dPressure );
   }
 
   template< int NUM_MAX_COMPONENTS >
   GEOSX_HOST_DEVICE
   void computeBodyForce( real64 const & solidDensity,
-                         real64 const & initialFluidTotalMassDensity,
                          real64 const & fluidTotalMassDensity,
                          real64 const & dFluidTotalMassDensity_dPressure,
                          real64 const ( &dFluidTotalMassDensity_dComponents)[NUM_MAX_COMPONENTS],
-                         real64 const & porosityInit,
                          real64 const & porosity,
                          real64 const & dPorosity_dVolStrain,
                          real64 const & dPorosity_dPressure,
                          real64 const ( &gravityVector )[3],
-                         real64 ( & bodyForceIncrement )[3],
+                         real64 ( & bodyForce )[3],
                          real64 ( & dBodyForce_dVolStrainIncrement )[3],
                          real64 ( & dBodyForce_dPressure )[3],
                          real64 ( & dBodyForce_dComponents )[3][NUM_MAX_COMPONENTS] ) const
   {
     computeBodyForce( solidDensity,
```

Comment:
`bodyForce` is induced by gravity only. How about tectonic stress/strain? Any plan for these?

Reply:
For stress/strain, the formulation that we have in place works well and does not require any change. For field-scale models, we still have to test the poromechanical initialization from a given regional stress, but my understanding is that we have everything we need now (I may be wrong :) ).